### PR TITLE
Documentation: added documentation for i.MX RT series and Teensy 4.x boards

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
This adds basic documentation (basic info + list of peripherals) for i. MX RT series (as discussed in #3462) and migrate Teensy 4.x documentation from boards/arm/imxrt/teensy-4.x to Documentation/ subfolder.

## Impact
Documentation only

### CODE REVIEW

File: Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+=================== ======================= ======
+
+Thus if the LED is statically on, NuttX has successfully booted and is,
+apparently, running normally. If the LED is flashing at approximately
+2Hz, then a fatal error has been detected and the system has halted.
+
+Configurations
+==============
+
+nsh-4.0:
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst:65]: remove `:` 
same thing on other subsections

File: Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Flash
+=====
+
+Teensy 4.x boards does not have debugger therefore external firmware has to be used to load NuttX.
+`Teensy Loader <https://www.pjrc.com/teensy/loader_cli.html>`_ can be installed and then NuttX can be loaded by
+
+    teensy_loader_cli --mcu=TEENSY41 -v -w nuttx.hex
+
+For Teensy 4.0 board, switch --mcu=TEENSY41 to --mcu=TEENSY40.
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst:141]:
```suggestion
.. code-block:: console
    $ teensy_loader_cli --mcu=TEENSY41 -v -w nuttx.hex

For Teensy 4.0 board, switch ``--mcu=TEENSY41`` to ``--mcu=TEENSY40``.
```

File: Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+That means for CAN1 -> can0, CAN2 -> can1 and CAN3 -> can2
+
+Bitrate and sample point can be also changed at System type peripheral selection,
+basic values are 1 MHz for bitrate and 0.80 for sample point. The FlexCAN driver
+for imxrt runs at 80 MHz clock frequency.
+
+The configuration also includes CAN utilities as candump and cansend.
+
+This configuration can be easily changed to work with Teensy 4.0 by
+selecting CONFIG_TEENSY_40=y.
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst:96]:
```suggestion
selecting ``CONFIG_TEENSY_40=y``.
```

File: Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+This configuration cannot be changed to Teensy 4.0 as this board does
+not have Ethernet capability.
+
+sd-4.1:
+-------
+
+This is an nsh configuration (see above) for Teensy-4.x with added support of
+connecting micro SD card.
+
+You can mount micro SD card by "mount -t vfat /dev/mmcsd0 /mnt"
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/boards/teensy-4.x/index.rst:117]:
```suggestion
You can mount micro SD card by:
.. code-block:: console
   $ mount -t vfat /dev/mmcsd0 /mnt
```

File: Documentation/platforms/arm/imxrt/index.rst [hunk 5]
```diff
@@ -0,0 +1,191 @@
+============
+i.MX RT
+============
+
+The i.MX RT series of chips from NXP Semiconductors is based around an ARM Cortex-M7 core running
+at 500 MHz, 600 MHz or 1 GHz based on particular MCUs
+
+Suported MCUs
+===================
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/index.rst:9]: The number underline characters needs to match the section name (same above)

**Review by v01d — CHANGES_REQUESTED** [review]
Minor changes

**Comment by michallenc** [comment]
@v01d $ mount -t vfat /dev/mmcsd0 /mnt seems to ignore that line completely, maybe change $ to #?

**Comment by v01d** [comment]
> @v01d $ mount -t vfat /dev/mmcsd0 /mnt seems to ignore that line completely, maybe change $ to #?

Maybe a blank line is needed between the code-block and the content.

**Comment by michallenc** [comment]
> > @v01d $ mount -t vfat /dev/mmcsd0 /mnt seems to ignore that line completely, maybe change $ to #?
> 
> Maybe a blank line is needed between the code-block and the content.

You´re right, that hepled. Thanks, all changes should be done.

File: Documentation/platforms/arm/imxrt/index.rst [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+A configurable module providing a range of functionality like emulation of a variety of
+serial/parallel communication protocols, flexible 16-bit timers or programmable logic blocks.
+This module is supported in i.MX RT1010 and higher.
+
+GPIO
+----
+
+Pins can be configured using ``imxrt_config_gpio()`` function. Writing to pins is
+done by ``imxrt_gpio_write()`` function and reading is done by ``imxrt_gpio_read()``.
```
Comment (v01d) [inline Documentation/platforms/arm/imxrt/index.rst:136]: Use :c:func: 
Not really critical now but it is good to have when we eventully add proper documentation for all these functions.

Comment (michallenc) [reply Documentation/platforms/arm/imxrt/index.rst]: Thanks. I forgot it there back then.

**Review by v01d — APPROVED** [review]

**Pull request merged**